src/rp2_common/hardware_irq/irq.c

```diff
@@ -410,3 +413,31 @@ void irq_init_priorities() {
     }
 #endif
 }
+
+#define FIRST_USER_IRQ (NUM_IRQS - NUM_USER_IRQS)
+
+static uint get_user_irq_claim_index(uint irq_num) {
+    invalid_params_if(IRQ, irq_num < FIRST_USER_IRQ || irq_num >= NUM_IRQS);
+    // we count backwards from the last, to match the existing hard coded uses of user IRQs in the SDK which were previously using 31
+    static_assert(NUM_IRQS - FIRST_USER_IRQ - 1 <= 8, ""); // we only use a single byte's worth of claim bits today.
```

Comment:
This looks to be off by one.

If number of IRQs is 32, and first user IRQ is 31, there is 1 user IRQ, but this expression would give 0.